flow/layers/default_layer_builder.cc

```diff
@@ -37,12 +37,13 @@ DefaultLayerBuilder::~DefaultLayerBuilder() = default;
 void DefaultLayerBuilder::PushTransform(const SkMatrix& sk_matrix) {
   SkMatrix inverse_sk_matrix;
   SkRect cullRect;
-  if (sk_matrix.invert(&inverse_sk_matrix)) {
+  // Perspective projections don't produce rectangles that are useful for
+  // culling for some reason.
+  if (sk_matrix.invert(&inverse_sk_matrix) && !inverse_sk_matrix.hasPerspective()) {
```

Comment:
Don't you mean if the sk_matrix has a perspective?

Reply:
Fixed